[#3332] docs: Axon Framework 5 - Getting Started Guide (#3369)
AxonFramework/AxonFramework: docs/af5-app-from-scratch/modules/ROOT/pages/implement-command-create-course.adoc

public class CreateCourseConfiguration {

    public static EventSourcingConfigurer configure(EventSourcingConfigurer configurer) {
        var stateEntity = EventSourcedEntityBuilder
                .annotatedEntity(CourseId.class, CreateCourseCommandHandler.State.class);  // <1>

        var commandHandlingModule = StatefulCommandHandlingModule.named("CreateCourse")
                .entities()
                .entity(stateEntity)  // <2>
                .commandHandlers()
                .annotatedCommandHandlingComponent(c -> new CreateCourseCommandHandler());
        return configurer.registerStatefulCommandHandlingModule(commandHandlingModule);
    }

}
----

<1> The `EventSourcedEntityBuilder` is a builder that allows us to create an entity with the given identifier type and state class.
We use `annotatedEntity`, because of the style we follow here, but you may also invoke just an `entity` builder method and do everything like defining event handlers here, so you ken keep your domain model free from annotations if it's your preferred way of coding.
<2> The `entity` method allows us to register the `State` class as the entity for the `CreateCourse` command handling module.

> With this step, we have completed the code to process a Command that represents the request to register a new bike in our system.

Author: Haha, or leave it as an Easter egg :)

<.> The `getComponent` method allows us to retrieve the `CommandGateway` component from the configuration.
<.> The `sendAndWait` method sends the command to the command bus and waits for the result.

NOTE: If you're using Spring Boot you can always define `NewConfiguration` as a `@Bean` and inject it into your controller. The first-class Spring support for Axon Framework 5 is under development.

NOTE: If you're familiar with Hexagonal Architecture (aka Ports & Adapters) you may treat the CommandGateway as a Port to your application and the controllers as Adapter.

== First in, first out

image::EventModeling_CreateCourse_Done.png[]

Do you like green tests?
What we like even more are green slices on Event Modeling.
So if you use this approach now you can mark your first slice as implemented!
Congratulations!

In the next section, we'll tackle a more complex feature: allowing students to subscribe to courses, where we have to deal with business rules spans wider scope of the system state.

== Alternative approach without annotations

> If you prefer not to use annotations in your domain model, you can go different way around.

Author: ```suggestion
If you prefer not to use annotations in your domain model, we have you covered.
```
Do we need the word `got` here :) ?
